CIScripts/Common/Invoke-CommandWithFunctions.ps1

```diff
@@ -0,0 +1,51 @@
+. $PSScriptRoot/Aliases.ps1
+function Invoke-CommandWithFunctions {
+    <#
+    .SYNOPSIS
+    This is a helper function for using locally defined functions in remote session.
+    The problem is that PowerShell does not support passing functions
+    to remote session, like ScriptBlocks or locally defined variables (by "$using:")
+    What Invoke-CommandWithFunctions do is just taking functions names and bodies 
+    and define them at remote session by using Invoke-Expression.
+    Then we invoke ScriptBlock with calls to already defined functions
+    After we remove the definitions from remote session memory so we do not pollute it.
+    .PARAMETER ScriptBlock
+    ScriptBlock with functions calls to invoke.
+    .PARAMETER Session
+    Remote session where the Scriptblock will be invoked.
+    .PARAMETER Functions
+    Names of functions called in ScriptBlock
```

Comment:
Names of locally defined functions to be made available in remote scope.* maybe?

Reply:
In theory, I don't have to call any of the "migrated" functions in the script block